qiskit/transpiler/passes/calibration/creators.py

```diff
@@ -0,0 +1,472 @@
+        self._config = backend.configuration()
+        self._channel_map = backend.configuration().qubit_channel_mapping
+
+    def supported(self, node_op: CircuitInst, qubits: List) -> bool:
+        """Determine if a given node supports the calibration.
+
+        Args:
+            node_op: Target instruction object.
+            qubits: Integer qubit indices to check.
+
+        Returns:
+            Return ``True`` is calibration can be provided.
+        """
+        return isinstance(node_op, RZXGate)
+
+    @staticmethod
+    def rescale_cr_inst(instruction: Play, theta: float, sample_mult: int = 16) -> Play:
+        """
+        Args:
+            instruction: The instruction from which to create a new shortened or lengthened pulse.
+            theta: desired angle, pi/2 is assumed to be the angle that the pulse in the given
+                play instruction implements.
+            sample_mult: All pulses must be a multiple of sample_mult.
+
+        Returns:
+            qiskit.pulse.Play: The play instruction with the stretched compressed
+                GaussianSquare pulse.
+
+        Raises:
+            QiskitError: if the pulses are not GaussianSquare.
+        """
+        pulse_ = instruction.pulse
+        if isinstance(pulse_, GaussianSquare):
+            amp = pulse_.amp
+            width = pulse_.width
+            sigma = pulse_.sigma
+            n_sigmas = (pulse_.duration - width) / sigma
+
+            # The error function is used because the Gaussian may have chopped tails.
+            gaussian_area = abs(amp) * sigma * np.sqrt(2 * np.pi) * math.erf(n_sigmas)
+            area = gaussian_area + abs(amp) * width
+
+            target_area = abs(theta) / (np.pi / 2.0) * area
+            sign = theta / abs(theta)
+
+            if target_area > gaussian_area:
+                width = (target_area - gaussian_area) / abs(amp)
+                duration = math.ceil((width + n_sigmas * sigma) / sample_mult) * sample_mult
+                return Play(
+                    GaussianSquare(amp=sign * amp, width=width, sigma=sigma, duration=duration),
+                    channel=instruction.channel,
+                )
+            else:
+                amp_scale = sign * target_area / gaussian_area
+                duration = math.ceil(n_sigmas * sigma / sample_mult) * sample_mult
+                return Play(
+                    GaussianSquare(amp=amp * amp_scale, width=0, sigma=sigma, duration=duration),
+                    channel=instruction.channel,
+                )
+        else:
+            raise QiskitError("RZXCalibrationBuilder only stretches/compresses GaussianSquare.")
+
+    def get_calibration(self, node_op: CircuitInst, qubits: List) -> Union[Schedule, ScheduleBlock]:
+        """Builds the calibration schedule for the RZXGate(theta) with echos.
+
+        Args:
+            node_op: Instruction of the RZXGate(theta). I.e. params[0] is theta.
```

Comment:
What happens if the input is not of type `RZXGate`?

Reply:
This will not happen since the method is only called after `supported` method returns `True`. In the `RZX` pass family, the supported method is equivalent to `isinstance(node_op, RZXGate)`.